src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = new Object[CAPACITY];
+        this.size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            enlargeCapacity();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == elementData.length) {
+            enlargeCapacity();
+        } else if (index > size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Incorrect index - " + index);
+        }
```

Comment:
why else if? they are 2 independent checks
index check is more reasonable to be put first